drivers/misc/stm32_tsc/stm32_tsc.c

```diff
@@ -0,0 +1,332 @@
+/*
+ * Copyright (c) 2024 Arif Balik <arifbalik@outlook.com>
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#define DT_DRV_COMPAT st_stm32_tsc
+
+#include <soc.h>
+#include <zephyr/irq.h>
+#include <zephyr/kernel.h>
+#include <zephyr/sys/util.h>
+#include <zephyr/devicetree.h>
+#include <zephyr/logging/log.h>
+#include <zephyr/drivers/reset.h>
+#include <zephyr/drivers/pinctrl.h>
+#include <zephyr/drivers/clock_control/stm32_clock_control.h>
+#include <zephyr/drivers/misc/stm32_tsc/stm32_tsc.h>
+
+/* each group only has 4 configurable I/O */
+#define GET_GROUP_BITS(val) (uint32_t)(((val) & 0x0f) << ((group->group - 1) * 4))
+
+LOG_MODULE_REGISTER(stm32_tsc, CONFIG_LOG_DEFAULT_LEVEL);
+
+struct stm32_tsc_group_config {
+	const uint8_t group;
+	const uint8_t channel_ios;
+	const uint8_t sampling_io;
+	const bool use_as_shield;
```

Comment:
do you need the `const` on all fields? the structure itself is going to be instantiated as constant so I think these are redundant, same below, just keep it for the pointers (in that case it says that it points to a constant value but the pointer itself is implicitly constant)